flake.nix

```diff
@@ -0,0 +1,339 @@
+
+                gmp
+                libgpg-error
+                wget
+                libconfig
+              ];
+
+              # preConfigure = ''
+              #   pushd standalone/src/main/resources/cz/crcs/ectester/standalone/libs/jni
+              #   make lib_timing.so
+              #   make lib_csignals.so
+              #   make lib_cppsignals.so
+              #   # make clean
+              #   make openssl
+              #   # openssl version
+              #   popd
+              # '';
+
+              # buildPhase = ''
+              #   ./gradlew clean build --offline
+              # '';
+
+              buildInputs = [
+                jdk17_headless
+                patched_openssl
+              ];
+
+              LD_LIBRARY_PATH = lib.makeLibraryPath [
+                libtommath
+                libtomcrypt
+                botan2
+                cryptopp
+                libgcrypt
+                patched_openssl
+                ninja
+                nettle
+                gmp
+                libgpg-error
+                libconfig
+              ];
+
+              # FIXME more things to copy here
+              installPhase = ''
+                mkdir -p $out
+                cp -r standalone/build $out
+                echo ${opensslVersion} > $out/build/opensslVersion
+              '';
+              
+              postFixup = ''
+
+                makeWrapper \
+                  ${jdk17_headless}/bin/java $out/bin/${pname} \
+                  --add-flags "-jar $out/build/libs/${pname}.jar" \
+                  --set LD_LIBRARY_PATH ${LD_LIBRARY_PATH}:$LD_LIBRARY_PATH
+              '';
+
+        });
+      in
+      {
+        packages = {
+          default = buildECTesterStandalone {
+            opensslVersion="3.3.1"; opensslHash="sha256-d3zVlihMiDN1oqehG/XSeG/FQTJV76sgxQ1v/m0CC34=";
+          };
+          openssl_331 = buildECTesterStandalone {
+            opensslVersion="3.3.1"; opensslHash="sha256-d3zVlihMiDN1oqehG/XSeG/FQTJV76sgxQ1v/m0CC34=";
+          }; 
+          openssl_322 = buildECTesterStandalone {
+            opensslVersion="3.2.2"; opensslHash="sha256-GXFJwY2enyksQ/BACsq6EuX1LKz+BQ89GZJ36nOOwuc=";
+          }; 
+          openssl_316 = buildECTesterStandalone {
+            opensslVersion="3.1.6"; opensslHash="sha256-XSvkA2tHjvPLCoVMqbNTByw6DibYpW+PCrn7btMtONc=";
+          }; 
+          openssl_3014 = buildECTesterStandalone {
+            opensslVersion="3.0.14"; opensslHash="sha256-7soDXU3U6E/CWEbZUtpil0hK+gZQpvhMaC453zpBI8o=";
+          }; 
+          # openssl_111w = buildECTesterStandalone "1.1.1w" "sha256-zzCYlQy02FOtlcCEHx+cbT3BAtzPys1SHZOSUgi3asg=";
+        };
+        devShells.default = with pkgs; mkShell rec {
+          buildInputs = [
+            # gradle2nix
+            gdb
+            ant
+            jdk17
+            pkg-config
+            global-platform-pro
+            gradle
+            # libraries to test
+            openssl_3013
```

Comment:
Is this supposed to be here?

Reply:
No, not really, but its debatable, which versions should be in the development shell - I suppose we'd use similar approach and have a default one and allow to specify others.